screens/ViewDetails/ViewAnnouncementScreen/ViewAnnouncementScreen.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { View, Text, Alert, TouchableOpacity } from 'react-native'
+import { RouteProp, useRoute } from '@react-navigation/native'
+import { Announcement } from '../../../types/Annoucements'
+import { styles } from './styles'
+import { viewDetailsStyles } from '../ViewDetailsStyles'
+import { globalStyles } from '../../../assets/global/globalStyles'
+import ProfilePicture from '../../../components/ProfilePicture/ProfilePicture'
+
+type RootStackParamList = {
+  ViewAnnouncement: {
+      announcement: Announcement;
+  }
+}
+
+type ViewAnnoucementScreenRouteProps = RouteProp<RootStackParamList, "ViewAnnouncement">
+
+const ViewAnnoucementScreen = () => {
+  const { announcement } = useRoute<ViewAnnoucementScreenRouteProps>().params
```

Comment:
I may be wrong, but same here, I do not think we need all of that for passing an Announcement as prop. I'd like to know your take on this.

Reply:
Same as above, it's to make typescript happy. You can try to go on this branch and do without that, you'll understand what I'm saying. There is maybe a simpler way to do it but I didn't manage to find one. The thing is you have to "register" the parameter in the navigation so the parameters can be "mapped" and you can get them back at arrival (plus you have to specify the type for typescript). If you find another way to do it tell me and we can change it.

Reply:
Thanks for the info. If it works, there is no worry to have I guess. I learnt swomething new today.🚀